Update ClimateEntity for horizontal swing support (#2309)
home-assistant/developers.home-assistant: docs/core/entity/climate.md

| preset_mode             | <code>str &#124; None</code>        | **Required by SUPPORT_PRESET_MODE**  | The current active preset.                                                 |
| preset_modes            | <code>list[str] &#124; None</code>  | **Required by SUPPORT_PRESET_MODE**  | The available presets.                                                     |
| swing_mode              | <code>str &#124; None</code>        | **Required by SUPPORT_SWING_MODE**   | The swing setting.                                                         |

> | swing_modes             | <code>list[str] &#124; None</code>  | **Required by SUPPORT_SWING_MODE**   | Returns the list of available swing modes, only vertial modes in the case horizontal swing is implemented. |

Reviewer: ```suggestion
| swing_modes             | <code>list[str] &#124; None</code>  | **Required by SUPPORT_SWING_MODE**   | Returns the list of available swing modes. Used exclusively for vertical modes if separate horizontal swing is implemented. |
```

| preset_modes            | <code>list[str] &#124; None</code>  | **Required by SUPPORT_PRESET_MODE**  | The available presets.                                                     |

> | swing_mode              | <code>str &#124; None</code>        | **Required by SUPPORT_SWING_MODE**   | The swing setting.                                                         |

Reviewer: maybe also update `swing_mode` description to state that if the entity provides horz swing, then this is used for vertical.

The device fan can have different swing modes that it wants the user to know about/control.

:::note

> In the case integrations don't have independent control of vertical and horizontal swing, then all possible options should be listed in `swing_modes`, otherwise `swing_modes` provides vertical support and `swing_horizontal_modes` should provide horizontal support.

Reviewer: ```suggestion
For integrations that don't have independent control of vertical and horizontal swing, all possible options should be listed in `swing_modes`, otherwise `swing_modes` provides vertical support and `swing_horizontal_modes` should provide horizontal support.
```
